### PR TITLE
Fix api collection icon rendering

### PR DESCRIPTION
**Pull request by RalucaP**
Bug/issue: rdar://135837611

## Summary

This PR addresses an issue where API Collections were displaying incorrect icons in external references. The issue occurred when external references to API Collections were rendering with `"role": "article"` instead of `"role": "collectionGroup"`, causing them to display article icons instead of the proper collection icons.

The implementation adds the missing `.collectionGroup` case to the `renderKindAndRole` function to ensure API Collections render consistently as articles with the collectionGroup role, displaying the correct collection icon in both main documents and external references.

## Dependencies

None.

## Testing

The functionality can be tested using the provided test case in `DocumentationContentRendererTests.swift`:

```swift
func testRenderKindAndRoleForAPICollection() throws {
    // API Collections should render as articles with collectionGroup role to display correct icon
    let (collectionKind, collectionRole) = DocumentationContentRenderer.renderKindAndRole(.collectionGroup, semantic: nil)
    XCTAssertEqual(collectionRole, "collectionGroup", "API Collections should have collectionGroup role for correct icon")
    
    // Verify other node types still work correctly
    let (articleKind, articleRole) = DocumentationContentRenderer.renderKindAndRole(.article, semantic: nil)
    XCTAssertEqual(articleKind, RenderNode.Kind.article)
    XCTAssertEqual(articleRole, "article")
    
    let (symbolKind, symbolRole) = DocumentationContentRenderer.renderKindAndRole(.class, semantic: nil)
    XCTAssertEqual(symbolKind, RenderNode.Kind.symbol)
    XCTAssertEqual(symbolRole, "symbol")
}
```

## Checklist
- [x] Updated tests
- [x] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

**Comment by RalucaP** [comment]
Hi Pat, I made some changes to address the radar concerns for cross-framework API collection icons, can you please take another look?

> There are many different DocumentationNode.Kind values... have we tested if any other icons are incorrect?

This radar is specific for API Collections, if more icons appear incorrect, we should fix them too, but probably in different PRs. :) Thanks!

File: Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift [hunk 1]
```diff
@@ -165,7 +165,7 @@ extension LinkDestinationSummary {
     
     /// Create a topic render render reference for this link summary and its content variants.
     func makeTopicRenderReference() -> TopicRenderReference {
-        let (kind, role) = DocumentationContentRenderer.renderKindAndRole(kind, semantic: nil)
+        let (kind, role) = DocumentationContentRenderer.renderKindAndRole(kind, semantic: nil, linkSummary: self)
```
Comment (patshaughnessy) [inline Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]:
```suggestion
        let (kind, role) = DocumentationContentRenderer.renderKindAndRole(kind, linkSummary: self)
```

You set the default value for `semantic` to be nil, so there's no need to pass that in here now.

File: Tests/SwiftDocCTests/Rendering/DocumentationContentRendererTests.swift [hunk 2]
```diff
@@ -131,6 +131,24 @@ class DocumentationContentRendererTests: XCTestCase {
             ]
         )
     }
+    
+    func testRenderKindAndRoleForAPICollection() throws {
```
Comment (patshaughnessy) [inline Tests/SwiftDocCTests/Rendering/DocumentationContentRendererTests.swift]: Test looks good, but you might want to add another test for the case when `taskGroups` is nil, i.e. test that this continues to work for references to a normal article.

And what should happen if `taskGroups` is an empty array?

**Review by patshaughnessy — APPROVED** [review]
Some minor suggestions. Thanks for the fix!

File: Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift [hunk 3]
```diff
@@ -270,6 +270,9 @@ public class DocumentationContentRenderer {
         default:
             if let article = semantic as? Article {
                 return (.article, roleForArticle(article, nodeKind: kind).rawValue)
+            } else if kind == .article, let summary = linkSummary, summary.taskGroups != nil {
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift]: The task groups of an link summary isn't a value that external documentation sources have to provide—and DocC doesn't include it in its link summaries unless specifically configured to do so—so it can't be reliably used to determine if an external page is an API collection or not.

File: Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift (hunk 1 above)
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: Question: Is there some other value about the link summary that we can use to determine if it's an API collection?

File: Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift [hunk 4]
```diff
@@ -247,7 +247,7 @@ public class DocumentationContentRenderer {
         return true
     }
 
-    static func renderKindAndRole(_ kind: DocumentationNode.Kind?, semantic: Semantic?) -> (RenderNode.Kind, String) {
+    static func renderKindAndRole(_ kind: DocumentationNode.Kind?, semantic: Semantic? = nil, linkSummary: LinkDestinationSummary? = nil) -> (RenderNode.Kind, String) {
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift]: If I understand this correctly, this method is not expected to receive both a semantic and a link summary because one represents a local page and the other represents an external page. If we need a version that accepts a link summary we could consider having either a separate overload (that calls into a private common implementation) or a parameter that can be either a Semantic or a LinkDestinationSummary.

Comment (sofiaromorales) [reply Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift]: If it's not expected to pass both semantic and linkSummary at the same invocation call, let's split the method into two making them an overload.

The original 
```swift
 static func renderKindAndRole(_ kind: DocumentationNode.Kind?, semantic: Semantic?) -> (RenderNode.Kind, String) {
```
and the overload
```swift
 static func renderKindAndRole(_ kind: DocumentationNode.Kind?, linkSummary: LinkDestinationSummary?) -> (RenderNode.Kind, String) {
```

Then the common logic can be encapsulated in another function.

File: Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift (hunk 1 above)
Comment (RalucaP) [reply Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: After analyzing the `LinkDestinationSummary` structure and the actual Apple documentation data, I found that there are no reliable alternatives to `taskGroups` for identifying API Collections in external references. The available fields (title patterns, URL patterns, abstract content, references) would all be fragile heuristics. 
We could add an optional `role` to `LinkDestinationSummary`, but that would be a much larger change. Would you prefer that?

Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: The `taskGroup` information isn't a reliable source for this information. It's not requited for documentation sources to include it and we are also [talking about removing it](https://github.com/swiftlang/swift-docc/pull/1311) because the link summary isn't meant to be a representation of the documentation hierarchy.

Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: Is there another `DocumentationNode.Kind` value that API collection pages should be using? I see that we have both `collection` and `collectionGroup` but I don't recall what either of those are meant to represent. If not, should we introduce a dedicated "API Collection" kind and use that?

Comment (patshaughnessy) [reply Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: I believe `collection` is the root/framework page, and `collectionGroup` are API Collections or pages that groups symbol pages together.

Comment (d-ronnqvist) [reply Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: Reading the kind values documentation I would think that a framework would use `module`. Regardless, if we have a kind value that represents API collections then it sounds like we should use that to identify API collections.

Comment (RalucaP) [reply Sources/SwiftDocC/Infrastructure/Link Resolution/ExternalPathHierarchyResolver.swift]: Thank you, both! I spoke with @d-ronnqvist and ended up taking a different approach, let me know your thoughts. :)

File: Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift (hunk 4 above)
Comment (RalucaP) [reply Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift]: Will keep this in mind for future implementation, thanks! For the time being, I took a different approach so this is not applicable anymore.

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift [hunk 5]
```diff
@@ -829,4 +829,51 @@ class LinkDestinationSummaryTests: XCTestCase {
...
+        let pageSummary = summaries[0]
+
+        XCTAssertEqual(pageSummary.kind, .collectionGroup)
+        XCTAssertEqual(pageSummary.title, "Time Pitch Algorithm Settings")
+        XCTAssertEqual(pageSummary.abstract, [.text("This is an API Collection that curates symbols.")])
+
+        // Verify round-trip encoding preserves the correct kind
+        let encoded = try JSONEncoder().encode(pageSummary)
+        let decoded = try JSONDecoder().decode(LinkDestinationSummary.self, from: encoded)
+        XCTAssertEqual(decoded.kind, .collectionGroup)
```
Comment (d-ronnqvist) [inline Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: minor: We have a test helper that does this
```suggestion
        try assertRoundTripCoding(pageSummary)
```

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift [hunk 6]
```diff
@@ -829,4 +829,51 @@ class LinkDestinationSummaryTests: XCTestCase {
...
+        let apiCollectionReference = ResolvedTopicReference(
+            bundleID: context.inputs.id,
+            path: "/documentation/TestBundle/APICollection",
+            sourceLanguage: .swift
+        )
+        let node = try context.entity(with: apiCollectionReference)
+        let renderNode = converter.convert(node)
+
+        let summaries = node.externallyLinkableElementSummaries(context: context, renderNode: renderNode)
+        let pageSummary = summaries[0]
```
Comment (d-ronnqvist) [inline Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: minor: If this page didn't produce any summaries, then line would trap and stop running all the other tests. If you use an optional expression that's wrapped in `XCTUnwrap`, then _if_ the code would behave unexpectedly, the failure would be limited to this test, and the rest would continue to run.
```suggestion
        let pageSummary = try XCTUnwrap(summaries.first)
```

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift [hunk 7]
```diff
@@ -829,4 +829,51 @@ class LinkDestinationSummaryTests: XCTestCase {
...
+    /// kind in linkable entities, ensuring cross-framework references display the correct icon.
+    func testAPICollectionKindForLinkDestinationSummary() async throws {
+        let symbolGraph = makeSymbolGraph(
+            moduleName: "TestModule",
+            symbols: [makeSymbol(id: "test-class", kind: .class, pathComponents: ["TestClass"])]
+        )
+
+        let catalogHierarchy = Folder(name: "unit-test.docc", content: [
+            TextFile(name: "APICollection.md", utf8Content: """
+                # Time Pitch Algorithm Settings
```
Comment (d-ronnqvist) [inline Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: minor: This title is oddly specific for a test
```suggestion
                # Some API collection
```

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift [hunk 8]
```diff
@@ -829,4 +829,51 @@ class LinkDestinationSummaryTests: XCTestCase {
...
+
+                This is an API Collection that curates symbols.
+
+                ## Topics
+
+                ### Algorithms
```
Comment (d-ronnqvist) [inline Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: minor: This topic name doesn't match the symbol that it's curating.

Because the API collection only has a single topic section, you can probably remove its title if we don't have a better title for it.
```suggestion
```

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift [hunk 9]
```diff
@@ -829,4 +829,51 @@ class LinkDestinationSummaryTests: XCTestCase {
...
+                - ``TestModule/TestClass``
+                """),
+            JSONFile(name: "TestModule.symbols.json", content: symbolGraph),
+            InfoPlist(displayName: "TestBundle", identifier: "com.test.example")
```
Comment (d-ronnqvist) [inline Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: Question: is it intentional that the Info.plist file specifies a name that's different from the module name? That's a bit unusual (but completely valid).

I'm asking because catalogs generally don't need an Info.plist file except that in this test you're hardcoding its name below when creating the reference, so removing it would cause the API collection to not be found for that reference string.

**Review by d-ronnqvist — COMMENTED** [review]
This looks like a good solution to me.

I'm curious how the code would behave if the API Collection specified an explicit "Article" kind using
```
@Metadata {
  @PageKind(article)
}
```

I feel that an explicit page kind like that should override any default kind inferred from the page's content. I think that _maybe_ the new code already works like this, but it would be good to have a second test that verifies that behavior.

(none of the individual questions or minor feedback are blocking)

**Comment by RalucaP** [comment]
Thanks for the thorough review, @d-ronnqvist ! All feedback addressed with improved test quality, better error handling, and added regression test for `@PageKind` precedence. :)
The fix maintains backward compatibility while correctly identifying API Collections in external references.

> I'm curious how the code would behave if the API Collection specified an explicit "Article" kind using
> 
> ```
> @Metadata {
>   @PageKind(article)
> }
> ```

Good catch! Added `testExplicitPageKindOverridesAPICollectionDetection()` to verify explicit `@PageKind(article)` takes precedence over automatic detection. The existing precedence logic already handles this correctly.

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift (hunk 9 above)
Comment (RalucaP) [reply Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: I believe this is correct! `'TestBundle' `(display name) vs `'TestModule'` (module name) should follow the convention for testing cross-module references.

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift (hunk 7 above)
Comment (RalucaP) [reply Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: Updated! Changed from specific 'Time Pitch Algorithm Settings' to generic 'API Collection' title to avoid confusion with real-world examples.

File: Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift (hunk 9 above)
Comment (d-ronnqvist) [reply Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: This test doesn't verify any cross-module linking behaviors and even then it's uncommon and a bit unconventional to specify a display name in the Info.plist that's _different_ from the module name. It's more common for documentation to either explicitly specify the same display name as the module name (often redundantly) or to not have an Info.plist at all.

Comment (RalucaP) [reply Tests/SwiftDocCTests/LinkTargets/LinkDestinationSummaryTests.swift]: Done! Removed the` Info.plist` from both test methods since they focus on API Collection kind detection rather than cross-module linking. Thanks!

**Review by d-ronnqvist — APPROVED** [review]

**Comment by d-ronnqvist** [comment]
@swift-ci please test
